server-rs/src/leagues.rs

```diff
@@ -75,8 +76,32 @@ pub struct League {
     pub signups_after: DateTime<Utc>,
     pub start_at: DateTime<Utc>,
     pub end_at: DateTime<Utc>,
-    pub badge_path: Option<String>,
-    pub image_path: Option<String>,
+    #[graphql(skip)]
+    pub badge_path_internal: Option<String>,
+    #[graphql(skip)]
+    pub image_path_internal: Option<String>,
     pub rules_and_info: Option<String>,
     pub link: Option<String>,
 }
+
+#[ComplexObject]
+impl League {
+    async fn image_path(&self, ctx: &async_graphql::Context<'_>) -> async_graphql::Result<Option<String>> {
+        let Some(image_path_internal) = &self.image_path_internal else {
```

Comment:
I think you need to embrace `match` and `Option.map` more but this is fine for now :P